src/field_theory/complete_ordered.lean

```diff
@@ -0,0 +1,594 @@
+    rcases h with ⟨⟨q, rfl⟩, h⟩,
+    exact le_of_lt h, },
+  { rintro w h,
+    obtain ⟨q2, hq2w, hq2q⟩ := exists_rat_btwn h,
+    use q2,
+    simp only [cut_image_rat, mem_image, exists_eq, mem_range, exists_and_distrib_right,
+      exists_eq_right, exists_prop_of_true, subtype.exists, rat.cast_inj, subtype.coe_mk,
+      subtype.val_eq_coe],
+    exact ⟨hq2q, hq2w⟩, },
+end
+
+lemma lt_induced_map_iff {F K : Type*} [linear_ordered_field F] [archimedean F]
+  [conditionally_complete_linear_ordered_field K] {x : F} {t : K} :
+  t < induced_map F K x ↔ ∃ (q : ℚ) (h : t < q), (q : F) < x :=
+begin
+  split,
+  { intro h,
+    obtain ⟨q, hqt, hqi⟩ := exists_rat_btwn h,
+    use [q, hqt],
+    rw induced_map at hqi,
+    by_contra hx,
+    push_neg at hx,
+    have hs := cSup_le_cSup (cut_image_bdd_above F K _)
+                (cut_image_nonempty F K x) (cut_image_subset F K hx),
+    change _ ≤ induced_map F K q at hs,
+    rw induced_map_rat at hs,
+    apply lt_irrefl (q : K) (lt_of_lt_of_le hqi hs), },
+  { rintro ⟨q, hqt, hqx⟩,
+    transitivity (q : K),
+    { exact hqt },
+    { obtain ⟨q2, hq2q, hq2x⟩ := exists_rat_btwn hqx,
+      rw induced_map,
+      have : (q2 : K) ∈ cut_image F K x := mem_cut_image_iff'.mpr hq2x,
+      apply lt_cSup_of_lt (cut_image_bdd_above F K x) this,
+      exact_mod_cast hq2q, }, },
+end
+
+lemma pointwise_add_Sup (A B : set F) (hA : A.nonempty) (hB : B.nonempty)
+  (hbA : bdd_above A) (hbB : bdd_above B) : Sup (A + B) = Sup A + Sup B :=
+begin
+  apply cSup_intro (nonempty.add hA hB),
+  { rintros f ⟨a, b, ha, hb, rfl⟩,
+    exact add_le_add (le_cSup _ _ hbA ha) (le_cSup _ _ hbB hb), },
+  { intros w hw,
+    have : w - Sup B < Sup A := sub_lt_iff_lt_add.mpr hw,
+    obtain ⟨a, ha, halt⟩ := exists_lt_of_lt_cSup hA this,
+    have : w - a < Sup B := sub_lt.mp halt,
+    obtain ⟨b, hb, hblt⟩ := exists_lt_of_lt_cSup hB this,
+    exact ⟨a + b, add_mem_add ha hb, sub_lt_iff_lt_add'.mp hblt⟩, }
+end
+
+lemma induced_map_add (F K : Type*) [linear_ordered_field F] [archimedean F]
+  [conditionally_complete_linear_ordered_field K] (x y : F) :
+  induced_map F K (x + y) = induced_map F K x + induced_map F K y :=
+begin
+  simp only [induced_map],
+  rw [cut_image_add, pointwise_add_Sup],
+  exact cut_image_nonempty F K x,
+  exact cut_image_nonempty F K y,
+  exact cut_image_bdd_above F K x,
+  exact cut_image_bdd_above F K y
+end
+
+/-- induced_map as an additive homomorphism -/
+def induced_add_map (F K : Type*) [linear_ordered_field F] [archimedean F]
+  [conditionally_complete_linear_ordered_field K] : F →+ K :=
+{ to_fun := induced_map F K,
+  map_zero' := by exact_mod_cast induced_map_rat F K 0,
+  map_add' := induced_map_add F K}
+
+/-- induced_map as a ring_hom -/
+def induced_add_mul_map (F K : Type*) [linear_ordered_field F] [archimedean F]
+  [conditionally_complete_linear_ordered_field K] :
+  F →+* K := ring_hom.mk_mul_self_of_two_ne_zero (induced_add_map F K)
+begin
```

Comment:
I think it makes sense to factor this `begin .. end` block into a separate lemma, so that the stuff that follows it isn't 3 screens away from the function that its an argument to (line above this comment).